### PR TITLE
Sound protocol V2 upgrade

### PR DESCRIPTION
**Pull request by gigamesh**

### CODE REVIEW

File: protocol/deploy/01_deploySound.ts [hunk 1]
```diff
@@ -115,67 +115,67 @@ const func: DeployFunction = async function ({ ethers, deployments, network, run
...
-    // for (const contract of contracts) {
-    //   // verify implementation
-    //   console.log('Verifying implementation:', contract.implementation);
-    //   await verifyContract(contract.implementation);
-    //   // verify proxy
-    //   console.log('Verifying proxy:', contract.proxy);
-    //   await verifyContract(contract.proxy, contract.args, contract.contractPath);
-    // }
-  }
+  // if (chainId !== 1337) {
```
Comment (vigneshka) [inline protocol/deploy/01_deploySound.ts:118]: what is this, and should we just delete this?

File: common/src/helpers.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
-import type { Provider } from '@ethersproject/providers';
```
Comment (vigneshka) [inline common/src/helpers.ts:1]: needed?

File: protocol/README.md [hunk 3]
```diff
@@ -1,16 +0,0 @@
-# sound.xyz protocol
```
Comment (vigneshka) [inline protocol/README.md:1]: We should have a readme

**Review by vigneshka — APPROVED** [review]

File: protocol/deploy/01_deploySound.ts (hunk 1 above)
Comment (gigamesh) [reply protocol/deploy/01_deploySound.ts]: its for verifying on etherscan. I just updated so we can turn it on via an env var

File: common/src/helpers.ts (hunk 2 above)
Comment (gigamesh) [reply common/src/helpers.ts:1]: fixed

File: package.json [hunk 4]
```diff
@@ -2,19 +2,25 @@
...
+      "protocol",
+      "common",
+      "bin-test"
+    ]
+  },
   "scripts": {
     "build": "turbo run build",
     "build:protocol": "turbo run build --scope=\"*/protocol\" --no-deps",
-    "ci:release": "pnpm -r publish --access public --no-git-checks",
+    "ci:release": "turbo run prepack && changeset publish --access public --no-git-checks && pnpm i --no-frozen-lockfile --lockfile-only --ignore-scripts",
```
Comment (PabloSzx) [inline package.json:15]: why did this change? this has to be pnpm -r publish, not changeset publish @gigamesh

Comment (PabloSzx) [reply package.json:15]: I guess this should have worked anyways and it seems like it did https://unpkg.com/@soundxyz/protocol@2.0.0/package.json

File: package.json [hunk 5]
```diff
@@ -2,19 +2,25 @@
   "name": "sound-xyz",
   "version": "0.1.0",
   "private": true,
+  "workspaces": {
+    "packages": [
+      "protocol",
+      "common",
+      "bin-test"
+    ]
+  },
```
Comment (PabloSzx) [inline package.json:11]: this was from yarn workspaces, which is not being used

File: protocol/README.md (hunk 3 above)
Comment (PabloSzx) [reply protocol/README.md:1]: was there a specific reason to remove this README?

**Review by PabloSzx — COMMENTED** [review]
I just created https://github.com/soundxyz/protocol-public/pull/20 to further improve this release

**Pull request merged**